src/main/java/at/hannibal2/skyhanni/data/mob/IslandExceptions.kt

```diff
@@ -38,200 +37,215 @@ object IslandExceptions {
-                    )
-                )
-
-                else -> null
-            }
-
-            IslandType.KUUDRA_ARENA -> when {
-                baseEntity is EntityMagmaCube && nextEntity is EntityMagmaCube -> MobData.MobResult.illegal
-                baseEntity is EntityZombie && nextEntity is EntityZombie -> MobData.MobResult.illegal
-                baseEntity is EntityZombie && nextEntity is EntityGiantZombie -> MobData.MobResult.illegal
-                else -> null
-            }
-
-            IslandType.WINTER -> when {
-                baseEntity is EntityMagmaCube && MobFilter.jerryMagmaCubePattern.matches(
-                    MobUtils.getArmorStand(
-                        baseEntity,
-                        2
-                    )?.name
-                ) ->
-                    MobData.MobResult.found(
-                        Mob(
-                            baseEntity,
-                            Mob.Type.BOSS,
-                            MobUtils.getArmorStand(baseEntity, 2),
-                            "Jerry Magma Cube"
-                        )
-                    )
-
-                else -> null
-            }
+        else -> null
+    }
 
-            else -> null
-        }
+    private fun deepCaverns(baseEntity: EntityLivingBase) = when {
+        baseEntity is EntityCreeper && baseEntity.baseMaxHealth.derpy() == 120 -> MobData.MobResult.found(
+            Mob(baseEntity, Mob.Type.BASIC, name = "Sneaky Creeper", levelOrTier = 3),
+        )
+
+        else -> null
+    }
+
+    private fun dwarvenMines(baseEntity: EntityLivingBase) = when {
+        baseEntity is EntityCreeper && baseEntity.baseMaxHealth.derpy() == 1_000_000 -> MobData.MobResult.found(
+            MobFactories.basic(baseEntity, "Ghost"),
+        )
+
+        else -> null
+    }
+
+    private fun crystalHollows(
+        baseEntity: EntityLivingBase,
+        armorStand: EntityArmorStand?,
+    ) = when {
+        baseEntity is EntityMagmaCube && armorStand != null && armorStand.cleanName() == "[Lv100] Bal ???❤" ->
+            MobData.MobResult.found(Mob(baseEntity, Mob.Type.BOSS, armorStand, "Bal", levelOrTier = 100))
+
+        else -> null
     }
 
+    private fun hub(
+        baseEntity: EntityLivingBase,
+        armorStand: EntityArmorStand?,
+        nextEntity: EntityLivingBase?,
+    ) = when {
+        baseEntity is EntityOcelot && armorStand?.isDefaultValue() == false &&
+            armorStand.name.startsWith("§8[§7Lv155§8] §cAzrael§r") ->
+            MobUtils.getArmorStand(baseEntity, 1).makeMobResult { MobFactories.basic(baseEntity, it) }
+
+        baseEntity is EntityOcelot && (nextEntity is EntityOcelot || nextEntity == null) -> MobUtils.getArmorStand(
+            baseEntity, 3,
+        ).makeMobResult { MobFactories.basic(baseEntity, it) }
+
+        baseEntity is EntityOtherPlayerMP &&
+            baseEntity.name.let { it == "Minos Champion" || it == "Minos Inquisitor" || it == "Minotaur " } &&
```

Comment:
Make this a pattern.
```
Mino(?:s Champion|s Inquisitor|taur )
```

Reply:
I dont like using a that complex regex here. rather use (x|y|z) for better readability